common/persistence/cassandra/nexus_service_store.go

```diff
@@ -61,30 +93,172 @@ func (s *NexusServiceStore) Close() {
+		return nil, gocql.ConvertError("GetNexusIncomingService", err)
+	}
+
+	return &p.InternalNexusIncomingService{
+		ServiceName:   serviceName,
+		NamespaceID:   namespaceID,
+		TaskQueueName: taskQueueName,
+		Metadata:      p.NewDataBlob(metadataBytes, metadataEncoding),
+		Version:       version,
+	}, nil
 }
 
 func (s *NexusServiceStore) ListNexusIncomingServices(
 	ctx context.Context,
 	request *p.InternalListNexusIncomingServicesRequest,
 ) (*p.InternalListNexusIncomingServicesResponse, error) {
-	return nil, serviceerror.NewUnimplemented("Cassandra ListNexusIncomingServices is not implemented")
-}
+	query := s.session.Query(templateListIncomingServicesQuery, constServicesPartition, rowTypeIncomingNexusService).WithContext(ctx)
+	iter := query.PageSize(request.PageSize).PageState(request.NextPageToken).Iter()
 
-func (s *NexusServiceStore) CreateOrUpdateNexusIncomingService(
-	ctx context.Context,
-	request *p.InternalCreateOrUpdateNexusIncomingServiceRequest,
-) error {
-	return serviceerror.NewUnimplemented("Cassandra CreateOrUpdateNexusIncomingService is not implemented")
+	response := &p.InternalListNexusIncomingServicesResponse{}
+	row := make(map[string]interface{})
+
+	for iter.MapScan(row) {
+		serviceName, err := getTypedFieldFromRow[string]("service_name", row)
+		if err != nil {
+			return nil, err
+		}
+		namespaceID, err := getTypedFieldFromRow[string]("namespace_id", row)
+		if err != nil {
+			return nil, err
+		}
+		taskQueueName, err := getTypedFieldFromRow[string]("task_queue_name", row)
+		if err != nil {
+			return nil, err
+		}
+		metadata, err := getTypedFieldFromRow[[]byte]("metadata", row)
+		if err != nil {
+			return nil, err
+		}
+		metadataEncoding, err := getTypedFieldFromRow[string]("metadata_encoding", row)
+		if err != nil {
+			return nil, err
+		}
+		version, err := getTypedFieldFromRow[int64]("version", row)
+		if err != nil {
+			return nil, err
+		}
+
+		response.Services = append(response.Services, p.InternalNexusIncomingService{
+			ServiceName:   serviceName,
+			NamespaceID:   namespaceID,
+			TaskQueueName: taskQueueName,
+			Metadata:      p.NewDataBlob(metadata, metadataEncoding),
+			Version:       version,
+		})
+
+		row = make(map[string]interface{})
+	}
+
+	if len(iter.PageState()) > 0 {
+		response.NextPageToken = iter.PageState()
+	}
+
+	if err := iter.Close(); err != nil {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("ListNexusIncomingServices operation failed. Error: %v", err))
+	}
+
+	return response, nil
 }
 
 func (s *NexusServiceStore) DeleteNexusIncomingService(
 	ctx context.Context,
-	name string,
+	serviceID string,
 ) error {
-	return serviceerror.NewUnimplemented("Cassandra DeleteNexusIncomingService is not implemented")
+	query := s.session.Query(templateDeleteIncomingServiceQuery,
+		constServicesPartition,
+		rowTypeIncomingNexusService,
+		serviceID,
+	).WithContext(ctx)
+
+	previous := make(map[string]interface{})
+	_, err := query.MapScanCAS(previous)
```

Comment:
Wondering why you need CAS here, is it just to know that the delete was applied?